### PR TITLE
ATF-7: Added Atomfeed config UI.

### PR DESCRIPTION
**Pull request by pkornowski**

### CODE REVIEW

File: omod/src/main/resources/config.xml [hunk 1]
```diff
@@ -15,7 +15,16 @@
 	<activator>org.openmrs.module.atomfeed.AtomfeedActivator</activator>
 
 	<require_version>2.0.5</require_version>
-	
+
+	<require_modules>
+		<require_module>
```
Comment (dserkowski) [inline omod/src/main/resources/config.xml]: We should require also a particular version of a module. Example: [link](https://github.com/openmrs/openmrs-module-referencemetadata/blob/3cfce170f944fb84cab4b03d6b33cbc52bbfb44c/omod/src/main/resources/config.xml#L21)

File: omod/src/main/resources/config.xml [hunk 2]
```diff
@@ -15,7 +15,16 @@
...
+			org.openmrs.module.uiframework
+		</require_module>
+		<require_module>
```
Comment (dserkowski) [inline omod/src/main/resources/config.xml]: As above

File: omod/pom.xml [hunk 3]
```diff
@@ -35,10 +35,48 @@
...
 			<scope>provided</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.openmrs.module</groupId>
+			<artifactId>uiframework-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.openmrs.module</groupId>
+			<artifactId>uicommons-omod</artifactId>
+			<scope>provided</scope>
```
Comment (dserkowski) [inline omod/pom.xml]: It is already defined in the main pom.xml file. Consider removing it.

File: omod/src/main/java/org/openmrs/module/atomfeed/page/controller/LoadFeedConfPageController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+public class LoadFeedConfPageController {
+
+	private static final String SAVE_CONFIG_ERROR = "atomfeed.configuration.json.save.fail";
+	private static final String SAVE_CONFIG_SUCCESS = "atomfeed.configuration.json.save.success";
+
+
+	protected static final Logger LOGGER = LoggerFactory.getLogger(LoadFeedConfPageController.class);
+
+	public String get(PageModel model) {
+		return "loadFeedConf";
```
Comment (dserkowski) [inline omod/src/main/java/org/openmrs/module/atomfeed/page/controller/LoadFeedConfPageController.java]: Use constant variable

File: pom.xml [hunk 5]
```diff
@@ -39,7 +39,23 @@
 		<module>omod</module>
 		<module>build-tools</module>
 	</modules>
-
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<groupId>org.openmrs.module</groupId>
+				<artifactId>uiframework-api</artifactId>
+				<version>3.2.2</version>
```
Comment (dserkowski) [inline pom.xml]: I try to run OpenMRS with this version of uiframework ([from there](https://github.com/openmrs/openmrs-module-uiframework/releases/tag/3.2.2)), but I got an exception. I propose using the latest version ([3.13.0](https://github.com/openmrs/openmrs-module-uiframework/releases/tag/3.13.0)). The exception: 
`Unable to start OpenMRS. Error thrown was: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'conversionService': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: java.util.Set org.openmrs.module.uiframework.UiFrameworkConversionServiceFactoryBean.springManagedConverters; nested exception is java.lang.TypeNotPresentException: Type org.openmrs.activelist.ProblemModifier not present
org.openmrs.module.ModuleException: Unable to start OpenMRS. Error thrown was: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'conversionService': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: java.util.Set org.openmrs.module.uiframework.UiFrameworkConversionServiceFactoryBean.springManagedConverters; nested exception is java.lang.TypeNotPresentException: Type org.openmrs.activelist.ProblemModifier not present
	at org.openmrs.web.WebDaemon.startOpenmrs(WebDaemon.java:65)
	at org.openmrs.web.Listener.contextInitialized(Listener.java:196)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:5077)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5591)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1574)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1564)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.servlet.ServletException: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'conversionService': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: java.util.Set org.openmrs.module.uiframework.UiFrameworkConversionServiceFactoryBean.springManagedConverters; nested exception is java.lang.TypeNotPresentException: Type org.openmrs.activelist.ProblemModifier not present
	at org.openmrs.web.Listener.startOpenmrs(Listener.java:276)
	at org.openmrs.web.WebDaemon$1.run(WebDaemon.java:42)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'conversionService': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: java.util.Set org.openmrs.module.uiframework.UiFrameworkConversionServiceFactoryBean.springManagedConverters; nested exception is java.lang.TypeNotPresentException: Type org.openmrs.activelist.ProblemModifier not present
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:334)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1202)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:537)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:741)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480)
	at org.openmrs.module.ModuleUtil.refreshApplicationContext(ModuleUtil.java:842)
	at org.openmrs.module.web.WebModuleUtil.refreshWAC(WebModuleUtil.java:866)
	at org.openmrs.web.Listener.performWebStartOfModules(Listener.java:658)
	at org.openmrs.web.Listener.performWebStartOfModules(Listener.java:637)
	at org.openmrs.web.Listener.startOpenmrs(Listener.java:268)
	... 1 more
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: java.util.Set org.openmrs.module.uiframework.UiFrameworkConversionServiceFactoryBean.springManagedConverters; nested exception is java.lang.TypeNotPresentException: Type org.openmrs.activelist.ProblemModifier not present
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:561)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:331)
	... 15 more
Caused by: java.lang.TypeNotPresentException: Type org.openmrs.activelist.ProblemModifier not present
	at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:117)
	at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
	at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at sun.reflect.generics.visitor.Reifier.reifyTypeArguments(Reifier.java:68)
	at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:138)
	at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at sun.reflect.generics.repository.ClassRepository.getSuperInterfaces(ClassRepository.java:108)
	at java.lang.Class.getGenericInterfaces(Class.java:913)
	at org.springframework.core.ResolvableType.getInterfaces(ResolvableType.java:399)
	at org.springframework.core.ResolvableType.as(ResolvableType.java:365)
	at org.springframework.core.ResolvableType.isAssignableFrom(ResolvableType.java:271)
	at org.springframework.core.ResolvableType.isAssignableFrom(ResolvableType.java:194)
	at org.springframework.beans.factory.support.GenericTypeAwareAutowireCandidateResolver.checkGenericTypeMatch(GenericTypeAwareAutowireCandidateResolver.java:113)
	at org.springframework.beans.factory.support.GenericTypeAwareAutowireCandidateResolver.isAutowireCandidate(GenericTypeAwareAutowireCandidateResolver.java:64)
	at org.springframework.beans.factory.annotation.QualifierAnnotationAutowireCandidateResolver.isAutowireCandidate(QualifierAnnotationAutowireCandidateResolver.java:142)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.isAutowireCandidate(DefaultListableBeanFactory.java:683)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.isAutowireCandidate(DefaultListableBeanFactory.java:642)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.isAutowireCandidate(DefaultListableBeanFactory.java:626)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1126)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1003)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:949)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:533)
	... 17 more
Caused by: java.lang.ClassNotFoundException: org.openmrs.activelist.ProblemModifier
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1856)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1705)
	at org.openmrs.module.ModuleClassLoader.loadClass(ModuleClassLoader.java:563)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
	... 38 more`

Comment (pkornowski) [reply pom.xml]: Okay, I'll look at this.

**Comment by dserkowski** [comment]
LGTM

**Pull request merged**